tools/pinning/current/pyproject.toml

```diff
@@ -63,10 +63,6 @@ mock = "*"
 # here in addition to certbot/setup.py because otherwise the pre-release
 # version of poetry will not be installed.
 poetry = ">=1.2.0a1"
-# setuptools-rust 1.0+ causes some of our tests to fail. In the long term, this
-# should be fixed, but this is a quick fix to get tests passing again. See
-# https://github.com/certbot/certbot/issues/9111 for more info.
-setuptools-rust = "<1"
```

Comment:
I also love it when problems fix themselves.

I worry we're setting ourselves up to be bit by this again though. [`setuptools-rust` is still a build dependency of cryptography](https://github.com/pyca/cryptography/blob/2720fa267477a9742ec8a9584d5f0f321ef9195e/pyproject.toml#L9), but it's no longer pinned in our requirements file because of the problem described in the second paragraph [here](https://github.com/certbot/certbot/blob/6c278c3352e84f14880f89021ed6fb9c62e1e320/tools/pinning/DESIGN.md#potential-problems).

What do you think about declaring a dependency on `setuptools-rust = "*"` with a comment like we have for `cython` above? (You could alternatively make a build dependency section of this file like we have [here](https://github.com/certbot/certbot/blob/6c278c3352e84f14880f89021ed6fb9c62e1e320/tools/pinning/oldest/pyproject.toml#L96-L104).) That way a version of `setuptools-rust` will be pinned in the requirements file and we'll have a little more stability in [scenarios where we have to build cryptography from source](https://dev.azure.com/certbot/certbot/_build/results?buildId=5096&view=logs&j=fdd3565a-f3c6-5154-eca9-9ae03666f7bd&t=5dbd9851-46a4-524f-73a8-4028241afcde&l=278).

If you wanted, we could do more work here by pinning [the build dependencies of setuptools-rust](https://github.com/PyO3/setuptools-rust/blob/main/pyproject.toml#L2) and [their build dependencies](https://github.com/pypa/setuptools_scm/blob/main/pyproject.toml#L3-L6) recursively. I know these packages would actually be used in places like the `test_sdist.sh` script which is what failed in https://github.com/certbot/certbot/issues/9111 because we build everything from source [there](https://github.com/certbot/certbot/blob/6c278c3352e84f14880f89021ed6fb9c62e1e320/letstest/scripts/test_sdists.sh#L49). I don't suspect this build dependency tree goes too much deeper, however, this work is somewhat annoying and probably not our highest priority since most of the Python ecosystem leaves build dependencies unpinned, so I'll leave it up to you to decide if you feel like doing this.